### DISCUSSION TITLE
NocoDB LXC - Config File / NC_Public_URL / NC_Backend_URL / NC_Dashboard_URL Settings to use with Reverse Proxy

### DISCUSSION THREAD

**Discussion by aeolius** [tteck/Proxmox#2662] [Help Needed] [answered]
Where can I edit the config for NocoDB or at least the three NC_PUBLIC_URL, NC_BACKEND_URL, NC_DASHBOARD_URL settings? I am having trouble getting NocoDB to work with my reverse proxy (HAProxy via PfSense) I can access it directly via IP but not via hostname setup in reverse proxy.

**Comment by tteck** [comment, marked as answer]
[Environment settings](https://github.com/nocodb/nocodb/blob/develop/packages/noco-docs/docs/020.getting-started/050.self-hosted/020.environment-variables.md) are absent, meaning it uses defaults. 

To set environment settings you can set them in the service file (`/etc/systemd/system/nocodb.service`)
Example:
To change the port to `8085`
```
[Unit]
Description=nocodb

[Service]
Type=simple
Restart=always
User=root
WorkingDirectory=/opt/nocodb
Environment="PORT=8085"
ExecStart=/opt/nocodb/./nocodb

[Install]
WantedBy=multi-user.target
```
then execute `systemctl daemon-reload` followed by `systemctl restart nocodb`

In your case you would need
`Environment="NC_PUBLIC_URL=xxxxxxxxxx"`
`Environment="NC_BACKEND_URL=xxxxxxxxxx"`
`Environment="NC_DASHBOARD_URL=xxxxxxxxxx"`
